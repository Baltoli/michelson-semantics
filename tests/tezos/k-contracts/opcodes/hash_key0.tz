$contract { parameter key;
storage (option key_hash);
code {CAR; HASH_KEY; SOME ;NIL operation; PAIR}
 } ;
$param "edpkuBknW28nW72KG6RoHtYW7p12T6GKc7nAbwYX5m8Wd9sDVC9yav" ;
$storage None ;
$ExpectedOutput (Some "tz1KqTpEZ7Yob7QbPE4Hy4Wo8fHG8LhKxZSx") ;

