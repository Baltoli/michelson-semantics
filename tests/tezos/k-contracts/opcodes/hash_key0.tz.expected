<generatedTop>
  <k>
    #VerifyStorage ( Some #KeyHash ( "tz1KqTpEZ7Yob7QbPE4Hy4Wo8fHG8LhKxZSx" ) )
  </k>
  <stack>
    Pair .List Some #KeyHash ( "edpkuBknW28nW72KG6RoHtYW7p12T6GKc7nAbwYX5m8Wd9sDVC9yav" )
  </stack>
  <paramtype>
    key .AnnotationList
  </paramtype>
  <storagetype>
    option .AnnotationList key_hash .AnnotationList
  </storagetype>
  <balance>
    #Mutez ( 0 )
  </balance>
  <amount>
    #Mutez ( 0 )
  </amount>
  <now>
    #Timestamp ( -1 )
  </now>
  <myaddr>
    #Address ( "InvalidMyAddr" )
  </myaddr>
  <knownaddrs>
    .Map
  </knownaddrs>
  <sourceaddr>
    #Address ( "InvalidSourceAddr" )
  </sourceaddr>
  <senderaddr>
    #Address ( "InvalidSenderAddr" )
  </senderaddr>
  <chainid>
    #ChainId ( 0 )
  </chainid>
</generatedTop>
